src/main/scala/pipelined/stage-register.scala

```diff
@@ -0,0 +1,64 @@
+// Definition for a pipeline stage register module and its interface
+
+// TODO: Change this to 'pipelined' package in a future patch
+package dinocpu
+
+
+import chisel3._
+import chisel3.internal._
+import chisel3.core.CompileOptions
+
+/** A [[Bundle]] which adds a `flush`, `bubble`, and `valid` bit to a data bundle.
+  * These values are observed by a StageReg during modification of its contents to
+  * either flush its contents to 0 when flush is high, ignore all writes when
+  * bubble is high, and write if `valid` is high.
```

Comment:
What's the difference between `bubble` and `valid`?

Can we call `valid` `write_enable` instead?

It seems to be that we just need a `write_enable` bit and the logic of `bubble || valid` should be in the pipelines.